### PR TITLE
feat: Deploy operator with che-namespace-editor clusterrole and and binding

### PR DESCRIPTION
**Pull request by sparkoo**
### What does this PR do?
Deploys che-operator with che-namespace-editor ClusterRole and ClusterRoleBinding for che-operator ServiceAccount.
It depends on che-operator PR https://github.com/eclipse/che-operator/pull/520

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/18272
fixes: https://github.com/eclipse/che/issues/18398

### How to test this PR?

deploy with che-operator from this PR https://github.com/eclipse/che/issues/18272 (or once it's merger into master) and check that che-operator ServiceAccount has cluster-wide permissions to update and get namespaces

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: src/tasks/installers/operator.ts [hunk 1]
```diff
@@ -88,6 +92,34 @@ export class OperatorTasks {
           }
         }
       },
+      {
+        title: `Create ClusterRole ${namespaceEditorclusterRoleName}`,
```
Comment (tolusha) [inline src/tasks/installers/operator.ts:96]: Pls, update deleteTasks as well.

Comment (sparkoo) [reply src/tasks/installers/operator.ts:96]: I did deleteTask and updateTask

**Comment by tolusha** [comment]
Since corresponding PR in the che-operator repository is merged pls update dependency by running `yarn` in the root of the project and commit yarn.lock file.

File: src/tasks/installers/operator.ts [hunk 2]
```diff
@@ -35,6 +37,8 @@ export class OperatorTasks {
   startTasks(flags: any, command: Command): Listr {
     const clusterRoleName = `${flags.chenamespace}-${this.operatorClusterRole}`
     const clusterRoleBindingName = `${flags.chenamespace}-${this.operatorClusterRoleBinding}`
+    const namespaceEditorclusterRoleName = `${flags.chenamespace}-${this.namespaceEditorClusterRole}`
+    const operatorNamespaceEditorclusterRoleBindingName = `${flags.chenamespace}-${this.operatorNamespaceEditorClusterRoleBinding}`
```
Comment (AndrienkoAleksandr) [inline src/tasks/installers/operator.ts:41]: minor:  operatorNamespaceEditorclusterRoleBindingName -> operatorNamespaceEditor**C**lusterRoleBindingName

File: src/tasks/installers/operator.ts [hunk 3]
```diff
@@ -201,6 +233,8 @@ export class OperatorTasks {
     const kube = new KubeHelper(flags)
     const clusterRoleName = `${flags.chenamespace}-${this.operatorClusterRole}`
     const clusterRoleBindingName = `${flags.chenamespace}-${this.operatorClusterRoleBinding}`
+    const namespaceEditorclusterRoleName = `${flags.chenamespace}-${this.namespaceEditorClusterRole}`
```
Comment (AndrienkoAleksandr) [inline src/tasks/installers/operator.ts:236]: namespaceEditorclusterRoleName -> namespaceEditor**C**lusterRoleName

File: src/tasks/installers/operator.ts (hunk 2 above)
Comment (sparkoo) [reply src/tasks/installers/operator.ts]: fixed

File: src/tasks/installers/operator.ts (hunk 3 above)
Comment (sparkoo) [reply src/tasks/installers/operator.ts]: fixed

**Comment by sparkoo** [comment]
> Since corresponding PR in the che-operator repository is merged pls update dependency by running `yarn` in the root of the project and commit yarn.lock file.

@tolusha: done

**Comment by tolusha** [comment]
@flacatus 
It seems CI is broken somehow
```
 See more details with DEBUG=*
        Error: ENOENT: no such file or directory, mkdir '/.config/chectl'
        Code: ENOENT
    { [Error: ENOENT: no such file or directory, mkdir] errno: -2, code: 'ENOENT', syscall: 'mkdir' }
```

**Comment by tolusha** [comment]
@sparkoo 
cherry-pick these changes.
They should help to pass tests.
https://github.com/che-incubator/chectl/pull/988/commits/554ec18cfd8468ad373ae5062cd70090ac3cd9c4

**Comment by sparkoo** [comment]
@tolusha I've rebased on master, the changes you've mentioned should be there.

**Comment by tolusha** [comment]
`workspace:inject --kubeconfig` is timed out.

**Comment by tolusha** [comment]
@sparkoo 
Could you pls increase command timeout to `60000` ?
https://github.com/che-incubator/chectl/blob/master/test/e2e/e2e.test.ts#L169

**Comment by sparkoo** [comment]
@tolusha done

**Review by tolusha — APPROVED** [review]
Small remarks. You have 3 places where `statusCode` is checked.
It is legacy code and we have already got rid of such checks because we have a meaningful error message anyway. 
```
  const statusCode = await kube.createClusterRoleFromFile(yamlFilePath, namespaceEditorClusterRoleName)
            if (statusCode === 403) {
    ...
            }
```
->
```
await kube.createClusterRoleFromFile(yamlFilePath, namespaceEditorClusterRoleName)
```

**Review by mmorhun — APPROVED** [review]

**Comment by sparkoo** [comment]
@tolusha I've removed the statusCode checks

**Review by tolusha — APPROVED** [review]

**Comment by tolusha** [comment]
Pls ignore `chectl e2e tests / Minikube Helm (pull_request)` it is know issue ^(

**Comment by sparkoo** [comment]
@tolusha yes, but now "Minikube Operator" is failing as well and it looks like it fails to prepare the environment :/

**Pull request merged**